receiver/factory.go

```diff
@@ -25,11 +26,11 @@ type TraceReceiverFactory interface {
 	// Type gets the type of the TraceReceiver created by this factory.
 	Type() string
 	// NewFromViper takes a viper.Viper config and creates a new TraceReceiver which uses next as the
-	// next TraceConsumer in the pipeline.
-	NewFromViper(cfg *viper.Viper, next consumer.TraceConsumer) (TraceReceiver, error)
-	// DefaultConfig returns the default configuration for TraceReceivers
+	// next TraceConsumer in the pipeline. Returning also the configuration used to create it.
+	NewFromViper(v *viper.Viper, next consumer.TraceConsumer) (receiver TraceReceiver, config interface{}, err error)
```

Comment:
Why do you need to return the configuration used?

Reply:
for logging

Reply:
Logging is another topic of unification, we use different logging systems (at least I saw that in some new PRs).

Can we instead get a logger as argument?

Reply:
will do